presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -673,7 +674,73 @@ public Node visitShowColumns(SqlBaseParser.ShowColumnsContext context)
     @Override
     public Node visitShowStats(SqlBaseParser.ShowStatsContext context)
     {
-        return new ShowStats(getLocation(context), getQualifiedName(context.qualifiedName()));
+        return new ShowStats(Optional.of(getLocation(context)), getQualifiedName(context.qualifiedName()));
+    }
+
+    @Override
+    public Node visitShowStatsParametrized(SqlBaseParser.ShowStatsParametrizedContext context)
+    {
+        // The following properties of SELECT subquery are required:
```

Comment:
These should not be parser checks. They should happen during analysis of the parsed tree.

Also, why aren't these supported, and why is the child not just a `query`?